RecoTracker/MkFit/plugins/MkFitOutputConverter.cc

```diff
@@ -175,6 +217,23 @@ void MkFitOutputConverter::produce(edm::StreamID iID, edm::Event& iEvent, const
   }
   const auto& mkFitGeom = iSetup.getData(mkFitGeomToken_);
 
+  // primary vertices under the algo_ because in initialStepPreSplitting there are no firstStepPrimaryVertices
+  // beamspot as well since the producer can be used in hlt
+  const reco::VertexCollection* vertices = nullptr;
+  const reco::BeamSpot* beamspot = nullptr;
+  if (algoCandSelection_) {
+    edm::Handle<reco::VertexCollection> hVtx;
+    iEvent.getByToken(verticesToken_, hVtx);
+    vertices = hVtx.product();
+    edm::Handle<reco::BeamSpot> hBs;
+    iEvent.getByToken(bsToken_, hBs);
+    beamspot = hBs.product();
```

Comment:
```suggestion
    beamspot = iEvent.get(bsToken_);
```

Reply:
this would require some more changes

either

```
  reco::VertexCollection vertices;
  reco::BeamSpot beamspot;
   if (algoCandSelection_) {
    vertices = iEvent.get(verticesToken_);
    beamspot = iEvent.get(bsToken_);
```

or 

```
  reco::VertexCollection* vertices = nullptr;
  reco::BeamSpot* beamspot = nullptr;
   if (algoCandSelection_) {
    *vertices = iEvent.get(verticesToken_);
    *beamspot = iEvent.get(bsToken_);
```

which way would be preferred?

Reply:
The first one would make copies (expensive in general), and the second one would result in undefined behavior because of trying to dereference `nullptr`. You could do
```cpp
  reco::VertexCollection* vertices = nullptr;
  reco::BeamSpot* beamspot = nullptr;
   if (algoCandSelection_) {
    vertices = &iEvent.get(verticesToken_);
    beamspot = &iEvent.get(bsToken_);
```

Reply:
ok, I take the third